galpy/potential/TwoPowerSphericalPotential.py

```diff
@@ -155,15 +185,119 @@ def _zforce(self,R,z,phi=0.,t=0.,_forceFloatEval=False):
         HISTORY:
            2010-07-09 - Written - Bovy (NYU)
         """
-        if not _forceFloatEval and not self.integerSelf == None:
+        if not _forceFloatEval and self.integerSelf is not None:
             return self.integerSelf._zforce(R,z,phi=phi,t=t)
-        else:
-            r= numpy.sqrt(R**2.+z**2.)
-            return -z/r**self.alpha*self.a**(self.alpha-3.)/(3.-self.alpha)\
-                *special.hyp2f1(3.-self.alpha,
-                                self.beta-self.alpha,
-                                4.-self.alpha,
-                                -r/self.a)
+        # else:
+        a, alpha, beta = self.a, self._alpha, self._beta
+        r= sqrt(R**2.+z**2.)
+        return -z/r**alpha*a**(alpha-3.)/(3.-alpha)\
+            *hyp2f1(3.-alpha, beta-alpha, 4.-alpha, -r/a)
+
+    # def _R2deriv(self,R,z,phi=0.,t=0.,_forceFloatEval=False):
+    #     """
+    #     NAME:
+    #         _R2deriv
+    #     PURPOSE:
+    #         evaluate the second radial derivative for this potential
+    #     INPUT:
+    #         R - Galactocentric cylindrical radius
+    #         z - vertical height
+    #         phi - azimuth
+    #         t- time
+    #     OUTPUT:
+    #         the second radial derivative
+    #     HISTORY:
+    #         2019-10-20 - Written - Starkman (UofT)
+    #     """
+    #     if not _forceFloatEval and self.integerSelf is not None:
+    #         return self.integerSelf._R2deriv(R,z,phi=phi,t=t)
+    #     # else:
+    #     a, alpha, beta = self.a, self.alpha, self.beta
+    #     r= sqrt(R**2.+z**2.)
+
+    #     if beta == 3.:
+    #         # Need to have derivatives of hyp2f1
+    #         raise ValueError('Not yet implemented')
+
+    #     if beta == 5.:
+    #         # Need to have derivatives of hyp2f1
+    #         raise ValueError('Not yet implemented')
+
+    #     return (Gamma(beta - 3) / r**5 *
+    #             (((z**2 - 2 * R**2) * Gamma(3 - alpha)) / Gamma(beta - alpha) +
+    #              a**(-3 + beta) * (R**2 + z**2)**(1 - beta / 2.) *
+    #              ((z**2 * (a * (alpha - beta) - r * (beta - 2)) +
+    #                R**2 * (r * (beta - 2) *
+    #                        (beta - 1) + 2 * a * (beta - alpha))
+    #                ) * hyp2f1(-2 + beta, beta - alpha + 1, beta,
+    #                           -(a / r)) / Gamma(beta) +
+    #               (a * (R**2 * (2 + alpha * (-3 + beta)) - z**2 * (beta - 2)) +
+    #                r * (-z**2 + R**2 * (beta - 1)) * (beta - 2)) *
+    #               (-2 + beta) * hyp2f1(beta - 1, beta - alpha + 1, beta,
+    #                                    -(a / r)) / Gamma(beta))))
```

Comment:
Not sure that we should keep this if it's going to be commented out...

Reply:
I've solved the special cases in Mathematica, but they involve partial derivatives of the hypergeometric function, which is not in scipy special functions.

Reply:
So maybe we should leave them out yet, if we ever wanted to add them it would be easy enough to re-do the Mathematica calc. (also, we can get 2nd derivatives with the ``NumericalDerivativesMixin``)

Reply:
deleted. Will (not) show up in next push.